Yuguang's solution for Yue z's project

yuguang003 opened the pull request:
@Yue-Z9 hi, hier is my solution, please check.
[solution](https://github.com/Dr-Eberle-Zentrum/Data-projects-with-R-and-GitHub/blob/YueZ/Projects/Yue_Z9/solution-from-Yuguang.md#chart-one)

martin-raden commented:
Hi Yuguang,
eventually, you should upload (beside the Rmd) the generated markdown ".md" file along with generated image files. So far, I find the markdown file missing.
Please upload as well and change the link in your first post.
Thanks,
Martin

martin-raden commented:
Furthermore, you have uploaded (along with the solution files within Yue's project) a lot of other files in other projects (in total 11 files!).

- [ ] Please cleanup and remove the files not part of the solution for Yue's project (and all files not within that folder).

THanks,
Martin

martin-raden commented on Projects/Yue_Z9/solution from Yuguang.Rmd:
```diff
@@ -0,0 +1,73 @@
+library(dplyr)
+library(ggplot2)
+library(lubridate)
+data <- read.csv("C:/Users/Yugua/Desktop/R2/Air_Quality.csv")
```
dont use absolute path information for reading files. this makes your code none-executable on other machines, since the path is specific to your system. instead:

- [ ] set the working directory to the source file's location
- [ ] read the file from the local directory without path specification.

that way, you code will be fine on any computer

martin-raden commented on Projects/Yue_Z9/solution from Yuguang.Rmd:
````diff
@@ -0,0 +1,73 @@
+```{r, echo=FALSE, message=FALSE}
+library(dplyr)
+library(ggplot2)
+library(lubridate)
````
it might be easier to load `library(tidyverse)` instead of specific sublibraries

martin-raden commented on Projects/Yue_Z9/solution from Yuguang.Rmd:
```diff
@@ -0,0 +1,73 @@
+  select(Name, Data.Value) |>
+  group_by(Name) |>
+  summarise(Mean_value = mean(Data.Value, na.rm = TRUE)) |>
+  mutate(Code = LETTERS[1:n()])
```
not sure if this is really the best solution to set up a dummy x-label.

instead I suggest to 
- [ ] use the "Name" for x 
- [ ] but disable showing the tick labels.

that way, you dont need an additional column without meaning.

martin-raden commented on Projects/Yue_Z9/solution from Yuguang.Rmd:
```diff
@@ -0,0 +1,73 @@
+    legend.position = "bottom",
+    legend.key.size = unit(0.1, "cm"),
+    legend.text = element_text(size = 6)
+  )
```
nice idea to show them below, but currently, some of the labels are outside of the plot.

- [ ] you might want to put the bar labels as a text annotation vertically onto each bar within the plot. 

that way, it is easier to identify what bar is representing which information. currently, it is partly hard to map based on the colors...

martin-raden commented on Projects/Yue_Z9/solution from Yuguang.Rmd:
```diff
@@ -0,0 +1,73 @@
+    Start_Date = ifelse(!grepl("^\\d{2}/\\d{2}/\\d{4}$", Start_Date), NA, Start_Date),
+    Year = year(dmy(Start_Date))) |>
+  arrange(desc(Data.Value)) |>
+  head(3) |>
```
since `head()` is a more general function, I recommend `slice_head()` in such a pipe

martin-raden commented on Projects/Yue_Z9/solution from Yuguang.Rmd:
```diff
@@ -0,0 +1,73 @@
+  print()
+#The second variable is to store the top 3 Place, according to the previous variable
+data_region <- data |>
+  filter(Name == "Asthma emergency department visits due to PM2.5") |>
```
- [ ] shouldnt this be based on the variable `data_pollutant`? 

avoid hard coded data information, to keep your coding general and able to process updated data!

martin-raden commented on Projects/Yue_Z9/solution from Yuguang.Rmd:
```diff
@@ -0,0 +1,73 @@
+#This variable is to select data we need according to previous two variables.
+data_task2 <- data |>
+  filter(Name == "Asthma emergency department visits due to PM2.5") |>
+  filter(Geo.Place.Name == c("Central Harlem - Morningside Heights", "East Harlem", "High Bridge - Morrisania")) |>
```
- [ ] both above lines should be based on the previously created variables...

martin-raden commented on Projects/Yue_Z9/solution from Yuguang.Rmd:
```diff
@@ -0,0 +1,73 @@
+  filter(Name == "Asthma emergency department visits due to PM2.5") |>
+  filter(Geo.Place.Name == c("Central Harlem - Morningside Heights", "East Harlem", "High Bridge - Morrisania")) |>
+  mutate(Year = year(dmy(Start_Date)))
+#visualization.
```
it seems the data is somewhat broken, since the visualization shows two values for the red line in 2005...

martin-raden commented on Projects/Yue_Z9/solution from Yuguang.Rmd:
```diff
@@ -0,0 +1,73 @@
+  filter(Geo.Place.Name == c("Central Harlem - Morningside Heights", "East Harlem", "High Bridge - Morrisania")) |>
+  mutate(Year = year(dmy(Start_Date)))
+#visualization.
+ggplot(data_task2, aes(x = Year, y = Data.Value, color = factor(Geo.Place.Name)))+
```
- [ ] ensure the year ticks are only on integers

martin-raden reviewed (commented):
Hi Yuguang,
beside the two comments from above concerning the files, here some comments on your coding assuming "Projects/Yue_Z9/solution from Yuguang.Rmd" is the final file..
Best,
Martin

yuguang003 commented:
Dear Martin and Yue,
i have modified my solution, please check, @martin-raden , and i'm sorry that i modified it so many times for testing, because i'm not sure what it looks like on Github.

Yue-Z9 commented:
Hi Yuguang,
For the first chart I would like you to put them in order from most to least so that it is clearer. Perhaps it could also xy-axis convert, which would add the name directly to the table, but of course, originally that would have been perfect. For the second chart it is better to add a title，so you know which pollution it is.  Other than that, everything looks perfect. 💯
